CRM/Contribute/BAO/Contribution.php

```diff
@@ -2468,6 +2468,9 @@ protected static function repeatTransaction(&$contribution, &$input, $contributi
       $createContribution = civicrm_api3('Contribution', 'create', $contributionParams);
       $contribution->id = $createContribution['id'];
       CRM_Contribute_BAO_ContributionRecur::copyCustomValues($contributionParams['contribution_recur_id'], $contribution->id);
+      if (!empty($input['membership_id'])) {
```

Comment:
This is  where the change is - it allows an input membership id  to  have this  effect

Reply:
Also, if we are going to add membership specific stuff into repeattransaction can we abstract it to it's own function as we know we have to do more refactoring/abstraction in this area eg. https://github.com/civicrm/civicrm-core/pull/14397